web/src/pages/Leaderboard.js

```diff
@@ -11,7 +11,7 @@ export default function Leaderboard() {
   const convertData = (obj, type, deltaObj) => {
     return Object.entries(obj).map(([key, value], index) => ({
       id: (index + 1).toString(),
-      name: key.replace(`${type}_`, ""),
+      name: key.includes("@") ? key.split("@")[0] : key.replace(`${type}_`, ""),
```

Comment:
what if user's name is `123@123`? i think we should probably implement email pattern matching.